src/Bicep.LangServer/Handlers/InsertResourceHandler.cs

```diff
@@ -80,7 +87,10 @@ public Task<Unit> Handle(InsertResourceParams request, CancellationToken cancell
                         Unit.Value);
                 }
 
-                var matchedType = azResourceTypeLoader.GetAvailableTypes()
+                var nsResolver = model.Binder.NamespaceResolver;
+                var azNamespace = nsResolver.GetNamespaceNames().Select(nsResolver.TryGetNamespace).First((ns) => ns?.ProviderName == AzNamespaceType.BuiltInName);
+                // Note(asilverman): I coerce azNamespace since there will always be an 'az' namespace defined in the compilation (either the builtin or the imported one). 
+                var matchedType = azNamespace!.ResourceTypeProvider.GetAvailableTypes()
```

Comment:
It's best to avoid using `!` in product code (tests are fine). Even when it's used safely, it's a place for future bugs to appear, as you're overriding the compiler's null analysis.

Something like this should avoid it:
```csharp
var nsResolver = model.Binder.NamespaceResolver;
var namespaces = nsResolver.GetNamespaceNames().Select(nsResolver.TryGetNamespace).WhereNotNull();
var azTypeProvider = namespaces.First(ns => ns.ProviderName == AzNamespaceType.BuiltInName).ResourceTypeProvider;

var matchedType = azTypeProvider.GetAvailableTypes()
```

Reply:
makes sense, thanks for showing me an alternative that is safer! TIL `WhereNotNull()`